### PR TITLE
Continue increasing buffer size

### PR DESCRIPTION
**Pull request by djamrozik**
When we are calling the `WriteString` method in `buffer.go`, we may need more space than double the current capacity. At the moment we only double capacity once and may not have enough space in the buffer. This pull request this will double until there is enough capacity and prevent the situation mentioned.

### CODE REVIEW

File: mmd/buffer.go [hunk 1]
```diff
@@ -128,7 +128,9 @@ func (b *Buffer) ensureSpace(sz int) {
 	if cap(b.data) > need {
 		return
 	}
-	b.doubleCapacity()
+	for cap(b.data) <= need {
+		b.doubleCapacity()
+	}
```
Comment (lquach) [inline mmd/buffer.go]: `if` block is redundant with this `for` loop.

File: mmd/buffer.go [hunk 2]
```diff
@@ -128,7 +128,9 @@ func (b *Buffer) ensureSpace(sz int) {
 	if cap(b.data) > need {
```
Comment (lquach) [inline mmd/buffer.go]: `if` block is redundant with this `for` loop.

Comment (djamrozik) [reply mmd/buffer.go]: changed, thanks

**Comment by lquach** [comment]
👍

**Comment by djamrozik** [comment]
Closing because change is included in #8

**Comment by djamrozik** [comment]
Closing because change is included in #8

**Pull request closed without merging**